### PR TITLE
PE-839: BATS for download folder

### PR DESCRIPTION
**Pull request by matibat**

### CODE REVIEW

**Review by fedellen — APPROVED** [review]

File: bats_test/downloads/folder-downloads/error_outputs.bats [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    cd "${DIR}"
+    run download_folder "${MY_FOLDER_ID}" "/non/existent/path"
+
+    [ "${status}" -eq 1 ]
+    [ "${output}" == "Error: ENOENT: no such file or directory, stat '/non/existent'" ]
+}
+
+@test "Errors out if the path exists as a non-file" {
+    cd "${DIR}"
+    mkfifo "${DIR}/some_fifo"
```
Comment (agsuy) [inline bats_test/downloads/folder-downloads/error_outputs.bats:27]: Should not this be "touch" ?

File: bats_test/downloads/folder-downloads/error_outputs.bats [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+@test "Errors out if the path is non-existent" {
+    cd "${DIR}"
+    run download_folder "${MY_FOLDER_ID}" "/non/existent/path"
+
+    [ "${status}" -eq 1 ]
+    [ "${output}" == "Error: ENOENT: no such file or directory, stat '/non/existent'" ]
+}
+
+@test "Errors out if the path exists as a non-file" {
```
Comment (agsuy) [inline bats_test/downloads/folder-downloads/error_outputs.bats:25]: Typo? Maybe ".. as a non-folder" ?

File: bats_test/downloads/folder-downloads/error_outputs.bats [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    [ "${output}" == "Error: The destination isn't a folder!" ]
+}
+
+@test 'Errors out if the folder ID is omitted' {
+    cd "${DIR}"
+    run download_folder
+
+    # FIXME: it should return error code 1
+    # [ "${status}" -eq 1 ]
+    [ "${output}" == "error: required option '-f --folder-id <folderId>' not specified" ]
```
Comment (agsuy) [inline bats_test/downloads/folder-downloads/error_outputs.bats:40]: Can we change these for assert_output?

https://github.com/ztombol/bats-assert#assert_output

File: bats_test/downloads/folder-downloads/error_outputs.bats [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+@test 'Errors out if the folder ID is invalid' {
+    INVALID_FOLDER_ID="edace8e-9d10-455f-ac47-25336fd3117b"
+    cd "${DIR}"
+    run -1 download_folder "${INVALID_FOLDER_ID}"
+
+    [ "${output}" == "Error: Invalid entity ID 'edace8e-9d10-455f-ac47-25336fd3117b'!'" ]
```
Comment (agsuy) [inline bats_test/downloads/folder-downloads/error_outputs.bats:48]: Maybe it is possible to parametrize entities?

e.g. 
[ "${output}" == "Error: Invalid entity ID '${INVALID_FOLDER_ID}'!'" ]

Comment (matibat) [reply bats_test/downloads/folder-downloads/error_outputs.bats]: Applied

File: bats_test/downloads/folder-downloads/error_outputs.bats (hunk 3 above)
Comment (matibat) [reply bats_test/downloads/folder-downloads/error_outputs.bats]: Applied

File: bats_test/downloads/folder-downloads/error_outputs.bats (hunk 1 above)
Comment (matibat) [reply bats_test/downloads/folder-downloads/error_outputs.bats]: You are right. Although the test as it were should be made for download file as it should error-out if the path is a non-file (e.g. a socket or a fifo)

**Review by fedellen — APPROVED** [review]

File: bats_test/downloads/folder-downloads/error_outputs.bats (hunk 1 above)
Comment (agsuy) [reply bats_test/downloads/folder-downloads/error_outputs.bats]: Agree

**Pull request merged**